programs/bpf_loader/src/lib.rs

```diff
@@ -162,12 +163,45 @@ pub fn process_instruction(
     instruction_data: &[u8],
     invoke_context: &mut dyn InvokeContext,
 ) -> Result<(), InstructionError> {
-    solana_logger::setup_with_default("solana=info");
-
+    process_instruction2(
+        program_id,
+        keyed_accounts,
+        instruction_data,
+        invoke_context,
+    )
+}
+#[allow(clippy::cognitive_complexity)]
+pub fn process_instruction2<'a>(
+    program_id: &Pubkey,
+    keyed_accounts: &'a [KeyedAccount],
+    instruction_data: &'a [u8],
+    invoke_context2: &'a mut dyn InvokeContext,
+) -> Result<(), InstructionError> {
     debug_assert!(bpf_loader::check_id(program_id));
 
+    let invoke_context = Rc::new(RefCell::new(invoke_context2));
+
+    // Route all log messages to the log collector
+    macro_rules! info {
```

Comment:
Put these macros inline so that we can do:
`info!("message");

Rather then:
`info!(invoke_context, "message");

Worth it?

Reply:
My gut feeling is this is a little too sneaky.  I think I favour an explicit `log_collector_info!()`-like macro for times when we want to communicate with the end-user

Reply:
hehe, sounds good